src/cluster/nodelabels.go

```diff
@@ -26,17 +26,25 @@ import (
 	"github.com/banzaicloud/pipeline/internal/cloudinfo"
 	"github.com/banzaicloud/pipeline/internal/global"
 	pipelineContext "github.com/banzaicloud/pipeline/internal/platform/context"
-	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
 	"github.com/banzaicloud/pipeline/pkg/common"
 )
 
 const labelFormatRegexp = "[^-A-Za-z0-9_.]"
 
+type NodePoolLabels struct {
+	Name         string
+	Existing     bool
+	InstanceType string            `json:"instanceType,omitempty"`
+	SpotPrice    string            `json:"spotPrice,omitempty"`
+	Preemptible  bool              `json:"preemptible,omitempty"`
+	Labels       map[string]string `json:"labels,omitempty"`
```

Comment:
I'd rename this to something like `CustomLabels` or `ExtraLabels` to make it clear that these are user provided labels while the others are set by Pipeline.

Reply:
renamed to CustomLabels